src/lib.rs

```diff
@@ -984,6 +984,35 @@ mod tests {
         }
     }
 
+    mod tuple_inputs {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn two_tuple() {
+            let StdoutTrimmed(output) = cmd!(("echo", "foo"));
+            assert_eq!(output, "foo");
+        }
+
+        #[test]
+        fn three_tuples() {
+            let StdoutTrimmed(output) = cmd!(("echo", "foo", "bar"));
+            assert_eq!(output, "foo bar");
+        }
+
+        #[test]
+        fn nested_tuples() {
+            let StdoutTrimmed(output) = cmd!(("echo", ("foo", "bar")));
+            assert_eq!(output, "foo bar");
+        }
+
+        #[test]
+        fn unit_input() {
+            let StdoutTrimmed(output) = cmd!(("echo", ()));
```

Comment:
It's an interesting consequence, maybe useful although I can't think of how immediately, that `()` as `Input` is a noop.

Reply:
You could imagine a wrapper function like this:
``` rust
fn run_docker<I: Input>(additional_args: I) {
  cmd_unit!(%"docker run", additional_args, "ubuntu:latest", "bash");
}
```
The function always performs the same `docker` command, but allows to tweak it by passing in additional arguments. And if callers want to not tweak anything, they can pass in `()`.

Reply:
Good example!